### PR TITLE
Return the updated ideal state for api after rebalance

### PR DESCRIPTION
**Pull request by npawar**

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/rebalance/DefaultRebalanceSegmentStrategy.java [hunk 1]
```diff
@@ -189,6 +191,10 @@ public IdealState apply(@Nullable IdealState idealState) {
         rebalanceIdealState(idealState, tableConfig, targetNumReplicas, rebalanceUserConfig,
             newPartitionAssignment);
         idealState.setReplicas(Integer.toString(targetNumReplicas));
+        for (Map.Entry<String, Map<String, String>> entry : idealState.getRecord().getMapFields().entrySet()) {
```
Comment (kishoreg) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/rebalance/DefaultRebalanceSegmentStrategy.java]: what if the update fails but the rebalanceIdealState is already updated. We should update the rebalanceIdealState only update to ZK was successful

Comment (npawar) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/rebalance/DefaultRebalanceSegmentStrategy.java]: HelixHelper.updateIdealState does not return any indication that it succeeded

Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/rebalance/DefaultRebalanceSegmentStrategy.java]: Can't we do all the updates to 'idealState' in the method rebalanceIdealState(...) and just return the updated idealState object?

Comment (npawar) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/rebalance/DefaultRebalanceSegmentStrategy.java]: That is exactly what is happening. But how do I send it back from rebalanceAndUpdateIdealState? It needs to be sent back to be displayed

Comment (npawar) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/rebalance/DefaultRebalanceSegmentStrategy.java]: That is exactly what is happening. But how do I send it back from rebalanceAndUpdateIdealState? It needs to be sent back to be displayed

Comment (snleee) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/rebalance/DefaultRebalanceSegmentStrategy.java]: Maybe we can create a blocking retry policy?

**Pull request merged**